### PR TITLE
Refactor code to fix bug and align with fixes

### LINKED ISSUE #59: Regex parsing bug
# Issue
Providing a field immediately after a parameter flag for the `catalog` command causes the quotes to be included in the entry. 
# Example/How to reproduce
This was done in `xenthm-branch-storage-coupling`, but should surface on `master` once it is merged by #53. 

```Wake up and slave~
catalog -a"bug"  
Successfully added author: "bug"
Oct 22, 2024 10:51:18 PM storage.Storage saveAuthorListToDataFile
INFO: Data saved
catalog -a"bug" -m"buggy"
Manga "buggy" added successfully to author "bug"
Oct 22, 2024 10:51:27 PM storage.Storage saveAuthorListToDataFile
INFO: Data saved
view
Here are the sla-I mean authors under you! Total: 3
1. test1
2. hello
3. "bug"
view -a"bug"
Mangas authored by "bug", Total: 1
1. "buggy" | Deadline: None
```
This is what the entry is saved as in `catalog.json`
image

### PR DESCRIPTION
**Pull request by averageandyyy**
Fixes #59

- Discoverd funky behavior of DefaultParser
  - DefaultParser strips quotations but not fully, implemented own stripQuotes method
  - DefaultParser registers -abug as a valid -a option, verified through debugger walkthrough which causes bug to be registerd as an argument. Implemented validateSpaceBetweenOptions method to ensure that flags/options are spaced-out and correctly parsed.
- Major refactoring from the use of String userInput and String[] userInputList. Previous iteration using userInput involved the use of a lot of redundant getUserInputAsList calls

### CODE REVIEW

File: src/test/java/parser/ParserTest.java [hunk 1]
```diff
@@ -89,32 +91,37 @@ public void getAuthorNameFromInput_validName_nameMatch() {
...
+
+//    //@@author averageandyyy
+//    @Test
+//    public void getMangaNameFromInput_invalidOption_parseExceptionThrown() {
+//        // Simulate a ParseException with an invalid option that leads to a TantouException
+//        String[] userInputList = {"-x", ""};
+//        assertThrows(TantouException.class, () -> {
+//            parser.getMangaNameFromInput(userInputList);
+//        });
+//    }
```
Comment (xenthm) [inline src/test/java/parser/ParserTest.java:126]: Reminder to remove commented line once they are resolved!

**Review by xenthm — APPROVED** [review]
Good job re-hauling the parser from the Apache Commons CLI to your own custom one! Just a minor nit regarding variable naming.

File: src/main/java/parser/Parser.java [hunk 2]
```diff
@@ -37,125 +36,24 @@ public Command getUserCommand(String userInput) throws TantouException {
...
-                    throw new TantouException("Invalid command provided!");
+        case BYE_COMMAND -> new ByeCommand();
+        case CATALOG_COMMAND ->
+                processCatalogCommand(trimmedUserInput);
+        case VIEW_COMMAND ->
+                processViewCommand(trimmedUserInput);
+        case SALES_COMMAND ->
+                processAddSalesCommand(trimmedUserInput);
+        default ->
+                throw new TantouException("Invalid command provided!");
```
Comment (xenthm) [inline src/main/java/parser/Parser.java:47]: I'm not sure I like how this is formatted. Perhaps store the result of the `switch` block first before returning it. This should help with potential formatting issues.

File: src/main/java/parser/Parser.java [hunk 3]
```diff
@@ -37,125 +36,24 @@ public Command getUserCommand(String userInput) throws TantouException {
...
         };
```
Comment (xenthm) [inline src/main/java/parser/Parser.java]: I'm not sure I like how this is formatted. Perhaps store the result of the `switch` block first before returning it. This should help with potential formatting issues.

**Review by sarahchow03 — APPROVED** [review]
Overall, the team needs to note the changes made in this particular PR due to the big change made to our parsing implementation. On this, good job on catching the library's bug early on and taking the effort to make this overhaul!

**Pull request merged**